src/cargo/util/toml/mod.rs

```diff
@@ -1965,11 +1965,10 @@ impl<P: ResolveToPath> DetailedTomlDependency<P> {
                 }
                 dep.set_artifact(artifact)
             } else {
-                let msg = format!(
+                bail!(
                     "`artifact = …` ignored as `-Z bindeps` is not set ({})",
```

Comment:
Now that this is an error rather than a warning, it should not just say "ignore". I would expect something like "`artifact = ...` requires `-Z bindeps`".

Reply:
🤦‍♂️ , thank you. It's fixed in c6ea1b7e35f8f5d78f601f105b536d38367d7445 .
I will try to be less rushed to cause less of these avoidable round-trips.